### DISCUSSION TITLE
[General] Got spammed by reth warning: WARN Hook is in progress, skipping forkchoice update. This may affect the performance of your node as a validator. hook=Prune

### DISCUSSION THREAD

**Discussion by ghost** [zkSync-Community-Hub/zksync-developers#691] [ZK Stack] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

I recently noticed that my L1 reth node logs are getting spammed with the same warning message over and over:
`2024-09-15T20:07:12.215979Z  WARN Hook is in progress, skipping forkchoice update. This may affect the performance of your node as a validator. hook=Prune head_block_hash=0xd5717783a4fa3f05e552a635cad6a9c90bb6de83fabe8d2254475e5f2efe0d4a safe_block_hash=0xd5717783a4fa3f05e552a635cad6a9c90bb6de83fabe8d2254475e5f2efe0d4a finalized_block_hash=0xd5717783a4fa3f05e552a635cad6a9c90bb6de83fabe8d2254475e5f2efe0d4a`
Has anyone else experienced this? Does this indicate a deeper issue, or is it just a benign log spam? I'm concerned about potential performance impacts on my node. Any insights or troubleshooting tips would be appreciated!
Thanks!

### Repo Link (Optional)

[ZK chains docker-compose](https://github.com/matter-labs/local-setup/blob/main/zk-chains-docker-compose.yml)

**Comment by alhenaai** [comment]
It seems like you are experiencing issues with your L1 reth node, specifically related to repeated warnings about skipping forkchoice updates due to an ongoing hook process. This warning indicates that the node is temporarily unable to update its view of the blockchain's state, which could potentially impact its performance as a validator.

Here are a few troubleshooting steps you might consider:

1. **Check Node Version**: Ensure that your node is running the latest version of the software. Sometimes, bugs related to process handling and performance are resolved in newer releases.

2. **Review Node Configuration**: Look into your node's configuration settings. There might be parameters related to the pruning process (`hook=Prune`) that you could adjust to prevent these warnings.

3. **Monitor System Resources**: Verify that your system has adequate resources (CPU, memory, disk I/O) to handle the node operations. Resource saturation can often lead to such issues.

4. **Consult Community Forums or Support**: If the issue persists, consider reaching out on relevant forums or directly to the support team of the software you are using. They might provide insights specific to known issues or configuration tweaks.

5. **Analyze Logs in Depth**: Look for any other related errors or warnings in the logs that occur before or after this message. This might give you more context on what might be causing the issue.

If these steps do not resolve the issue, it might be beneficial to provide more detailed logs and system information to a technical support team for a deeper analysis.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Reply by ghost** [reply]
1. I use v0.2.0-beta.2, which has been referenced in original compose config file
2. Again, I'm using stock config, here's comand for your reference:   `command: node --dev --datadir /rethdata --http --http.corsdomain "*" --http.addr 0.0.0.0 --http.port 8545  --dev.block-time 300ms --chain /chaindata/reth_config`
3. I am running it on Google Cloud's VM, n2d-standard-8 instance

**Reply by alhenaai** [reply]
Given the details you've provided, it seems you are running a relatively recent version of the software and using a robust VM setup. Here are a few additional suggestions to further investigate and potentially resolve the issue:

1. **Update to the Latest Version**: Even though you are using a recent beta version, check if there are any newer versions or patches available that address performance issues or bugs related to the pruning process.

2. **Adjust Block Time**: The `--dev.block-time 300ms` setting is quite aggressive and could be contributing to the performance issues, especially if the node is under heavy load or interacting with complex transactions. Consider increasing the block time to see if it alleviates the warning messages.

3. **Optimize HTTP Settings**: The current HTTP settings (`--http.addr 0.0.0.0 --http.port 8545`) are open, which is typical for development but can be adjusted if you're experiencing performance issues. Ensure that your node isn't handling unwanted external requests that could be impacting performance.

4. **Review and Optimize Chain Configuration**: The `--chain /chaindata/reth_config` might have configurations that are too resource-intensive or not optimized for your specific use case. Review the chain configuration for any settings that could be adjusted to improve performance.

5. **Monitor and Analyze Performance Metrics**: Since you are running on a Google Cloud VM, utilize Google Cloud's monitoring tools to track CPU, memory, and disk I/O. Look for any anomalies or resource saturation around the times you see the warnings in the logs.

6. **Consult with the Software Maintainers**: If the issue persists, it might be helpful to reach out directly to the maintainers of the software with detailed logs and your configuration. They might have more insights or suggest configuration changes specific to your environment.

By following these steps, you should be able to better diagnose the issue and potentially mitigate the performance warnings you are encountering.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by porteradams12** [comment]
Adding more details shared in tg:
Reth logs: [https://app.warp.dev/block/ZRxh4RBXovVyc19YS1Et1T](url)
Reth.toml: [https://pastebin.com/UW08LFVc](url)

**Reply by ghost** [reply]
reth config from similar issue:
https://github.com/paradigmxyz/reth/issues/7500#issuecomment-2088025087
I'm trying to switch the reth version in the docker image from v0.2.0-beta.2, to the latest tag now, let's see if this will help

**Reply by bxpana** [reply]
Keep us posted on how that goes.

**Reply by ghost** [reply, marked as answer]
I will continue monitoring reth over the next couple of weeks. Keep in mind, this issue occurred after running Reth for about a week or so.

**Reply by ghost** [reply]
I confirm that reth is stable now, I don't see this WARN message again in the logs.

**Reply by bxpana** [reply]
That's great to hear, glad everything is running as expected now 😁
